app/models/invenio_rdm_record_converter.rb

```diff
@@ -286,4 +290,75 @@ def version(content)
 
     "v#{version_number}.0.0"
   end
+
+  def format_publication_date(publication_date)
+    normalize_date(publication_date)
+  end
+
+  def normalize_date(date_string)
```

Comment:
This is a beast of a function and, as you can probably tell, it's brittle. This is the best I can come up with given how many cases we need to cover. I don't see a reason to refactor right now since this is probably only going to be used for the migration.